src/logging.h

```diff
@@ -7,14 +7,19 @@
 
 #include <libopencm3/stm32/usart.h>
 
+#include "clock.h"
+
 enum { LOG_LEVEL_INFO, LOG_LEVEL_WARNING, LOG_LEVEL_ERROR };
 static const char *const log_level_strings[] = {"INFO", "WARNING", "ERROR"};
 
 int _write(int file, char *ptr, int len);
 
 #define LOG_MESSAGE(level, format, arg...)                                     \
-	printf("%s,%s:%d," format "\n", log_level_strings[level], __FILE__,    \
-	       __LINE__, ##arg)
+	do {                                                                   \
```

Comment:
I don't like the function inside the macro but I have not a better idea so OK for me.

Reply:
Agree... :joy::sweat_smile: